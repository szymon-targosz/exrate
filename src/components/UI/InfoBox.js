import React from 'react';
import PropTypes from 'prop-types';

const InfoBox = ({ description, data, cssClasses, ...rest }) => {
   let classes = [ 'info-box' ];

   if (cssClasses) classes = classes.concat(cssClasses);

   return (
      <div className={classes.join(' ')} {...rest}>
         <p className="info-box__description">{description}</p>
         <h4>{data}</h4>
      </div>
   );
};

InfoBox.propTypes = {
   cssClasses: PropTypes.oneOfType([ PropTypes.string, PropTypes.array ]),
   description: PropTypes.string.isRequired,
   data: PropTypes.string.isRequired
};

export default InfoBox;
