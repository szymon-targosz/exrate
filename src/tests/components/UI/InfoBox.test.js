import React from 'react';
import { shallow } from 'enzyme';
import InfoBox from '../../../components/UI/InfoBox';

let wrapper;

beforeEach(() => {
   wrapper = shallow(<InfoBox description="Some description" data="19932.31" />);
});

describe('COMPONENT - InfoBox', () => {
   test('Should render InfoBox correctly', () => {
      expect(wrapper).toMatchSnapshot();
   });

   test('Should add external css classes - array', () => {
      wrapper.setProps({ cssClasses: [ 'bg-secondary', 'border', 'border-info' ] });
      expect(wrapper).toMatchSnapshot();
   });

   test('Should add external css classes - string', () => {
      wrapper.setProps({ cssClasses: 'bg-warning p-s' });
      expect(wrapper).toMatchSnapshot();
   });
});
