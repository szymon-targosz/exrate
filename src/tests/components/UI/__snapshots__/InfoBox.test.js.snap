// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - InfoBox Should add external css classes - array 1`] = `
<div
  className="info-box bg-secondary border border-info"
>
  <p
    className="info-box__description"
  >
    Some description
  </p>
  <h4>
    19932.31
  </h4>
</div>
`;

exports[`COMPONENT - InfoBox Should add external css classes - string 1`] = `
<div
  className="info-box bg-warning p-s"
>
  <p
    className="info-box__description"
  >
    Some description
  </p>
  <h4>
    19932.31
  </h4>
</div>
`;

exports[`COMPONENT - InfoBox Should render InfoBox correctly 1`] = `
<div
  className="info-box"
>
  <p
    className="info-box__description"
  >
    Some description
  </p>
  <h4>
    19932.31
  </h4>
</div>
`;
